### DISCUSSION TITLE
how to bind async method in PropertyChangedCallback of styled property?

### DISCUSSION THREAD

**Discussion by chenjing1294** [AvaloniaUI/Avalonia#11461] [Q&A]
in WPF:
```c#
public static readonly DependencyProperty MytextDependencyProperty = DependencyProperty.Register("MyText", typeof(string), typeof(TestClass), new PropertyMetadata(null, PropertyChangedCallbackAsync));

private static async void PropertyChangedCallbackAsync(DependencyObject d, DependencyPropertyChangedEventArgs e)
{
    await someMethod();
}
```

**Comment by maxkatz6** [comment]
There is no difference if it's async or sync. WPF executes your code like it's a regular sync void. It does not await anything.
In the case of Avalonia there also won't be a difference between sync or async. If you wish, subscribe on property changes as normal and add "async void" there. For instance, OnPropertyChanged method override.
